### PR TITLE
Fix #16 - Dommy refactoring with better performance

### LINKED ISSUE #16: nextSibling issue
```js
  let parent = new Dommy()
  const childNodes = [];
  for (let i = 0; i < 100; i++) childNodes.push(new Nody(parent, i));

  console.log(childNodes[0].nextSibling)  // here it's end node
  parent.appendChild(childNodes[1])
  console.log(childNodes[0].nextSibling) // here it's childNodes[1]
```

### PR DESCRIPTION
**Pull request by WebReflection**
As discussed in https://github.com/luwes/js-diff-benchmark/issues/16, the current Dommy implementation might not satisfy most libraries, or work as expected.

This MR uses a more DOM friendly API, enabling possible usage of real DOM, or better DOM based libraries, plus it adds _Prepend_ benchmark, in which _Snabdom_ seems to constantly fail, while _Spect_ randomly fails on the random benchmark.

Other libraries seems to be OK.

P.S. failing libraries should not be placed on top of the table, imho, but we need to wrap `console.assert` in order to spot which library failed where, or at least flag the whole cell as red/invalid.

### CODE REVIEW

File: src/bench.js [hunk 1]
```diff
@@ -227,63 +239,118 @@ table.sort((a, b) => {
...
+
+function random(parent, diff, oldNodes) {
   return diff(
     parent,
-    parent.childNodes,
-    Array.from(parent.childNodes).reverse(),
-    parent.lastElementChild
+    oldNodes,
+    shuffleSeed.map((newIdx) => oldNodes[newIdx]),
+    before
```
Comment (dy) [inline src/bench.js:302]: This `before` is not passed as argument - is that needed here?

**Comment by WebReflection** [comment]
### Current Status

Note that _snabdom_ failed.

Screenshot from 2020-04-17 14-13-31

File: src/bench.js (hunk 1 above)
Comment (WebReflection) [reply src/bench.js:341]: It's on top of the file hence needed in every diff call. You can remove the comment and see it used

**Comment by luwes** [comment]
I'll merge, can fix Snabbdom later

**Pull request merged**